out/cloud_foundry.go

```diff
@@ -43,6 +46,22 @@ func (cf *CloudFoundry) Target(organization string, space string) error {
 	return cf.cf("target", "-o", organization, "-s", space).Run()
 }
 
+func (cf *CloudFoundry) PushAppWithRollingDeployment(
+	path string,
+	currentAppName string,
+	dockerUser string,
+	showLogs bool,
+	noStart bool,
+	manifest string,
+) error {
+	cf.cf("version").Run()
+
+	if manifest != "" {
+		cf.ApplyManifest(currentAppName, manifest)
```

Comment:
Does this order of operations require the app to already exist at least once before a rolling update can be performed? Since it appears in order to do `v3-apply-manifest <app-name>` you have to already have the application in place?

Reply:
I think this PR has sat long enough that a version of this resource could be built using the v7 `cf cli`. It has a better UX than the alpha `v3-` prefixed commands in the v6 CLI and is tolerant to the app not yet existing.

These docs demonstrate that workflow:
https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html